src/ai/susi/server/api/cms/GetSkillsByAuthor.java

```diff
@@ -78,8 +55,15 @@ public ServiceResponse serviceImpl(Query call, HttpServletResponse response, Aut
             json.put("message","Author name not given");
             return new ServiceResponse(json);
         }
-        final File folder = new File(DAO.model_watch_dir.getPath());
-        listFilesForFolder(author.toLowerCase(),folder);
+
+        for (Map.Entry<SusiSkill.ID, SusiSkill> entry : DAO.susi.getSkillMetadata().entrySet()) {
+            SusiSkill.ID skill = entry.getKey();
+            if ((entry.getValue().getAuthor()!=null) && entry.getValue().getAuthor().toLowerCase().contains(author.toLowerCase())) {
```

Comment:
I like this implementation but if `.toLowerCase()` is being standardly used throughout the repo, it makes a lot more sense to have `getAuthor()` method `return whatever-its-returning.toLowerCase()`. Is there a reason for the explicit lowercase check here?

Reply:
`getAuthor()` returns the author name written in the skill.txt file, Since the user might write in different cases that's why I converted it to lowercase before comparing. Yes, It makes more sense to return the result of getAuthor in lowercase and in skill_cms we can capitalize the author name before display.